File: 8-string-to-integer-atoi/string-to-integer-atoi.java
class Solution {
    public int myAtoi(String s) {
        s = s.trim();
        int a = 0; 
        int sign = 1;
        long out=0;
        while(a != s.length()){
            if(s.charAt(a) == '-' && a == 0){
                sign = -1;
            }else if(s.charAt(a) == '+' && a == 0){
                sign = 1;
            }else if(Character.isDigit(s.charAt(a))) {
                int temp = s.charAt(a) - '0';
                out=out*10+temp;
                if(sign == 1 && out > Integer.MAX_VALUE){
                    return Integer.MAX_VALUE;
                }else if(sign == -1 && -out < Integer.MIN_VALUE){
                    return Integer.MIN_VALUE;
                }
                
            }else{
                break;
            }
            
            a++;
        }
        
        return (int)(sign * out);
        
    }
}

File: 8-string-to-integer-atoi/string-to-integer-atoi.cpp
class Solution {
public:
    string ans = "";
    int sign = 1;
    long long converttoINT(string s){
        long long ans = 0;
        for(int i=0; i<s.size(); i++){
            char ch = s[i]-'0';
            ans=ans*10+(int)ch;

            if(sign > 0 && ans>=INT_MAX){
                return INT_MAX;
            }else if(sign<0 && sign*ans<=INT_MIN){
                return INT_MIN;
            }
        }

        return ans;
    }

    string Atoi(string s, int i){        
        if(i == s.size()){
            return ans;
        }
        if(i==0 && s[i] == '-'){
            sign = -1;
        }else if(i==0 && s[i] == '+'){
            sign = 1;
        }else if (s[i] >= '0' && s[i] <= '9'){
            ans+=s[i];
        }else{
            return ans;
        }

        return Atoi(s, i+1);
    }

    int myAtoi(string s) {
        int t = 0;
        for(int k=0; k<s.size(); k++){
            if(s[k] == ' '){
                t++;
            }else{
                break;
            }
        }

        string newS = s.substr(t);

        string out = Atoi(newS, 0);
        if(out == "") return 0;

        long long a = converttoINT(out);
        
        if(sign < 0) a=-a;

        return a;
    }
};